kcidb/db/__init__.py

```diff
@@ -134,6 +135,24 @@ def empty(self):
         assert self.is_initialized()
         self.driver.empty()
 
+    def reset(self, database=None):
+        """
+        Recreates a database driver instance on request
+        using existing database string
+        Args:
+            database:   A string specifying the database to access, formatted
+                        as "<DRIVER>:<PARAMS>" or just "<DRIVER>". Where
+                        "<DRIVER>" is the driver name, and "<PARAMS>" is the
+                        optional driver-specific database parameter string
+                        or None.
```

Comment:
The method shouldn't accept a new database string, as its whole purpose is to recreate the client for the existing database. If the user wants to connect to another database, they should just create a new client. Please remove the parameter.